x/community/keeper/incentives.go

```diff
@@ -51,7 +58,85 @@ func (k Keeper) StartDisableInflationUpgrade(ctx sdk.Context) {
 
 	logger.Info("x/kavadist inflation disabled")
 
-	// todo: consolidate community funds (transfer from kavadist and community pool)
-
 	logger.Info("disable inflation upgrade finished successfully!")
 }
+
+// StartCommunityFundConsolidation consolidates the community funds from
+// x/distribution and x/kavadist into the x/community module account
+func (k Keeper) StartCommunityFundConsolidation(ctx sdk.Context) error {
+	logger := k.Logger(ctx)
+	logger.Info("community fund consolidation upgrade started")
+
+	// Consolidate x/distribution community pool
+	if err := k.consolidateCommunityDistribution(ctx); err != nil {
+		return err
+	}
+
+	// Consolidate x/kavadist account
+	if err := k.consolidateCommunityKavadist(ctx); err != nil {
+		return err
+	}
+
+	// Log new x/community balance
+	communityCoins := k.GetModuleAccountBalance(ctx)
+	logger.Info(fmt.Sprintf("community funds consolidated, x/community balance is now %s", communityCoins))
+
+	return nil
+}
+
+// consolidateCommunityDistribution transfers all coins from the x/distribution
+// community pool to the x/community module account
+func (k Keeper) consolidateCommunityDistribution(ctx sdk.Context) error {
+	logger := k.Logger(ctx)
+
+	// Get community coins with leftover leftoverDust
+	truncatedCoins, leftoverDust := k.distrKeeper.
+		GetFeePoolCommunityCoins(ctx).
+		TruncateDecimal()
+
+	// Transfer to x/community
+	err := k.bankKeeper.SendCoinsFromModuleToModule(
+		ctx,
+		distrtypes.ModuleName, // sender
+		types.ModuleName,      // recipient
+		truncatedCoins,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to transfer x/distribution coins to x/community: %w", err)
+	}
+
+	logger.Info(fmt.Sprintf("transferred %s from x/distribution to x/community", truncatedCoins))
+
+	// Set x/distribution community pool to remaining dust amounts
+	k.distrKeeper.SetFeePool(ctx, distrtypes.FeePool{
+		CommunityPool: leftoverDust,
+	})
+
+	logger.Info("updated x/distribution community pool to 0")
```

Comment:
this shouldn't be exact 0 because of dust. Perhaps print out the dust amount here?